diff --git a/mkosi/sandbox.py b/mkosi/sandbox.py
index 83433720..5d254031 100755
--- a/mkosi/sandbox.py
+++ b/mkosi/sandbox.py
@@ -85,7 +85,7 @@ class cap_user_data_t(ctypes.Structure):
     ]
 
 
-libc = ctypes.CDLL(None, use_errno=True)
+libc = ctypes.CDLL("@LIBC@", use_errno=True)
 
 libc.syscall.restype = ctypes.c_long
 libc.unshare.argtypes = (ctypes.c_int,)
@@ -186,7 +186,7 @@ def seccomp_suppress_chown() -> None:
     owned by their own uid. To still allow non-root users to build images, if requested we install
     a seccomp filter that makes calls to chown() and friends a noop.
     """
-    libseccomp = ctypes.CDLL("libseccomp.so.2")
+    libseccomp = ctypes.CDLL("@LIBSECCOMP@")
     if libseccomp is None:
         raise FileNotFoundError("libseccomp.so.2")
 
