diff --git a/mkosi/qemu.py b/mkosi/qemu.py
index 0793a60f..eb2e9463 100644
--- a/mkosi/qemu.py
+++ b/mkosi/qemu.py
@@ -182,7 +182,7 @@ def find_ovmf_firmware(config: Config, firmware: QemuFirmware) -> Optional[OvmfC
     if not firmware.is_uefi():
         return None
 
-    desc = list((config.tools() / "usr/share/qemu/firmware").glob("*"))
+    desc = list((config.tools() / "@QEMU_FIRMWARE@").glob("*"))
     if config.tools() == Path("/"):
         desc += list((config.tools() / "etc/qemu/firmware").glob("*"))
 
